bigquery/google/cloud/bigquery/table.py

```diff
@@ -816,22 +843,41 @@ def table_type(self):
         """
         return self._properties.get('type')
 
+    @property
+    def time_partitioning(self):
+        """google.cloud.bigquery.table.TimePartitioning: Configures time-based
+        partitioning for a table.
+        """
+        prop = self._properties.get('timePartitioning')
+        if prop is not None:
+            return TimePartitioning.from_api_repr(prop)
+
     @property
     def partitioning_type(self):
         """Union[str, None]: Time partitioning of the table if it is
         partitioned (Defaults to :data:`None`).
-
-        The only partitioning type that is currently supported is ``'DAY'``.
         """
-        return self._properties.get('timePartitioning', {}).get('type')
+        warnings.warn(
+            "This method will be deprecated in future versions. Please use "
+            "TableListItem.time_partitioning.partitioning_type instead.",
+            PendingDeprecationWarning)
+        if self.time_partitioning is not None:
+            return self.time_partitioning.partitioning_type
 
     @property
     def partition_expiration(self):
         """Union[int, None]: Expiration time in milliseconds for a partition.
+
+        If this property is set and :attr:`partitioning_type` is not set,
```

Comment:
Looks like this comment about `partitioning_type` migrated to the wrong property.

Reply:
I think it is in the appropriate doc string. It's about something that will happen to `partitioning_type` based on how you set `partition_expiration`. The behavior existed previously, so I thought I would call out the unique defaulting that is happening. I will change "this property" to "partition_expiration" to make it more clear.